packages/foam-vscode/src/features/orphans.ts

```diff
@@ -0,0 +1,130 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { Foam, Note } from 'foam-core';
+import { getOrphansConfig, OrphansConfig } from '../settings';
+import { FoamFeature } from '../types';
+
+const feature: FoamFeature = {
+  activate: async (
+    context: vscode.ExtensionContext,
+    foamPromise: Promise<Foam>
+  ) => {
+    const foam = await foamPromise;
+    const config = getOrphansConfig();
+    const provider = new OrphansProvider(foam, config);
+
+    context.subscriptions.push(
+      vscode.window.registerTreeDataProvider('foam-vscode.orphans', provider)
+    );
+
+    foam.notes.onDidUpdateNote(() => provider.refresh());
+  },
+};
+
+export default feature;
+
+class OrphansProvider implements vscode.TreeDataProvider<OrphanTreeItem> {
+  // prettier-ignore
+  private _onDidChangeTreeData: vscode.EventEmitter<OrphanTreeItem | undefined | void> = new vscode.EventEmitter<OrphanTreeItem | undefined | void>();
+  // prettier-ignore
+  readonly onDidChangeTreeData: vscode.Event<OrphanTreeItem | undefined | void> = this._onDidChangeTreeData.event;
+
+  private orphans: Note[] = [];
+  private exclude: string[] = [];
+  private root = vscode.workspace.workspaceFolders[0].uri.fsPath;
+
+  constructor(private foam: Foam, private config: OrphansConfig) {
+    this.exclude = config.exclude.map(d => path.normalize(`/${d}`));
+    this.computeOrphans();
+  }
+
+  refresh(): void {
+    this.computeOrphans();
+    this._onDidChangeTreeData.fire();
+  }
+
+  private computeOrphans() {
+    this.orphans = this.foam.notes
+      .getNotes()
+      .filter(note => !this.foam.notes.getAllLinks(note.uri).length)
+      .filter(note => {
+        const p = note.uri.path.replace(this.root, '');
```

Comment:
I noticed that `root` uses `fsPath`, whereas here you use `path`, might be better to use `path` across the board?